bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java

```diff
@@ -0,0 +1,350 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hdpowerview.internal.database;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class containing the database of all known shade 'types' and their respective 'capabilities'.
+ *
+ * If user systems detect shade types that are not in the database, then this class can issue logger warning messages
+ * indicating such absence, and prompting the user to report it to developers so that the database and the respective
+ * binding functionality can (hopefully) be extended over time.
+ *
+ * @author Andrew Fiddian-Green - Initial Contribution
+ */
+@NonNullByDefault
+public class ShadeCapabilitiesDatabase {
+
+    private final Logger logger = LoggerFactory.getLogger(ShadeCapabilitiesDatabase.class);
+
+    /*
+     * Database of known shade capabilities.
+     */
+    private static final Capabilities[] CAPABILITIES_DATABASE = {
```

Comment:
Consider if using a `Map` or something similar would make sense for _O(1)_ lookup in getCapabilities/isCapabilitiesInDatabase.

Reply:
Ok. Love the very "hairy" code..

```
private static final Map<Integer, Capabilities> CAPABILITIES_DATABASE = Arrays.asList(
    new Capabilities(0).primary().tiltOnClosed().text("Bottom Up"),
    ... etc.
    new Capabilities()).stream().collect(Collectors.toMap(Capabilities::getValue, Function.identity()));
```